build.assets/Dockerfile-arm

```
@@ -1,8 +1,75 @@
-ARG BUILDBOX_VERSION
-FROM public.ecr.aws/gravitational/teleport-buildbox:$BUILDBOX_VERSION
+# This Dockerfile is used to build Teleport on ARM only.
+# We are using the official Debian 11 image as a base image
+# because the final binary must be compatible with distroless
+# images that are also Debian 11 based: https://github.com/GoogleContainerTools/distroless
+FROM docker.io/library/debian:11
 
-USER root
+COPY locale.gen /etc/locale.gen
+COPY profile /etc/profile
+
+ENV LANGUAGE="en_US.UTF-8" \
+    LANG="en_US.UTF-8" \
+    LC_ALL="en_US.UTF-8" \
+    LC_CTYPE="en_US.UTF-8" \
+    DEBIAN_FRONTEND="noninteractive"
+
+# BUILDARCH is automatically set by DOCKER when building the image with Build Kit (MacOS by default).
+# https://docs.docker.com/engine/reference/builder/#automatic-platform-args-in-the-global-scope
+ARG BUILDARCH
 
 RUN apt-get -y update && \
-    apt-get -y install gcc-arm-linux-gnueabihf gcc-aarch64-linux-gnu && \
-    apt-get -y autoclean && apt-get -y clean
+    apt-get install -q -y --no-install-recommends \
+        build-essential \
+        ca-certificates \
+        curl \
+        git \
+        gzip \
+        libc6-dev \
+        libpam-dev \
+        locales \
+        pkg-config \
+        sudo \
+        unzip \
+        zip \
+        # ARM dependencies
+        gcc-arm-linux-gnueabihf \
+        libc6-dev-armhf-cross \
+        && \
+    dpkg-reconfigure locales && \
+    apt-get -y clean && \
+    rm -rf /var/lib/apt/lists/*
+
+# Install Node.js.
+ARG NODE_VERSION
+ENV NODE_PATH="/usr/local/lib/nodejs-linux"
+ENV PATH="$PATH:${NODE_PATH}/bin"
+RUN NODE_ARCH=$(if [ "$BUILDARCH" = 'amd64' ]; then echo 'x64'; else echo 'arm64'; fi) && \
```

I don't see why this would cause problems:

```suggestion
RUN NODE_ARCH="$(if [ "$BUILDARCH" = 'amd64' ]; then echo 'x64'; else echo 'arm64'; fi)" && \
```

Seems to work fine in my clone of your branch.

---

I missed that. The outer quotes are working, only the inner ones are problematic :sweat_smile: